src/core/reference/include/openvino/reference/grid_sample.hpp

```diff
@@ -20,10 +21,20 @@ namespace {
 using index_4D_t = typename std::array<size_t, 4>;
 
 template <typename GRID_ET>
-using denormalize_fn_t = typename std::function<GRID_ET(GRID_ET, size_t)>;
+using denormalize_fn_t = GRID_ET (*)(GRID_ET, size_t);
```

Comment:
Is the change required? IMO `std::function` a little clearer to understand than *

Reply:
Agree that `std::function` has nicer syntax and much more flexibility but features provided by `std::function` are not used here.

The main reason of change is for binary size reduction and also to remove overhead it can produce (should be minimal when storing function pointer), and e.g. when `get_padded_fn` is set by `switch case` first is default initialized then overwritten, using function pointer is uninitialized and then just set to correct address (as fundamental type). 

Just done some binary size tests and differences between two version is around 3KiB (function pointer gives smaller size), but this assume as only one type is used in evaluate. When expanding this evaluates by adding `i64` for two stages the difference increased to around 12 KiB, with `std::function` binary size can grow much faster.